deepspeed/runtime/engine.py

```diff
@@ -574,6 +574,8 @@ def _configure_distributed_model(self, model):
         self.module = model
         if self.fp16_enabled():
             self.module.half()
```

Comment:
Need to add an assert here to prevent case where a Z3 module is initialized as fp32 with zero.init. Simply doing a self.module.half is not going to work here for zero.init initialized module.

Reply:
Also need to avoid clipping for fp32 in engine for stage 2 and stage 3

Reply:
Clipping for fp32 is now avoided in engine for ZeRO